lib/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -1165,6 +1156,7 @@ public void onMessage(ITransport transport, ProtocolMessage message) throws Ably
                     }
                     break;
                 case connected:
+                    disconnectedRetryAttempt = 0;
```

Comment:
Curious as to why this has been added here, rather than keeping what we previously had in `setState` - would we not want to base the attempt on the state the library enters, rather than a protocol message received (what happens if the state changes under the hood in the meantime)?

This also looks like a change in behaviour from what was previously here - in the past we would also set the retry attempt to 0 if the channel entered the suspended state?

Reply:
This is a good point. 
Although I had given a thought on this one and it seems once you receive connected message, state will destined to go into `connected` state and after `suspended` state can never go into `disconnected` state directly. It has to go through connected state first. So, adding check for `suspended` is actually redundant. 

I will see if I can make changes for this.

Reply:
done